### PR TITLE
fix tests that were failing via RPC

### PR DESCRIPTION
**Pull request by winsvega**
Here is the first test filled with retesteth via RPC and  custom cpp client version. 
this test should be compatible with existing cpp-develop.

### CODE REVIEW

File: GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json [hunk 1]
```diff
@@ -2,10 +2,11 @@
     "createNameRegistratorPerTxsNotEnoughGas" : {
         "_info" : {
             "comment" : "",
-            "filledwith" : "cpp-1.3.0+commit.619a6438.Linux.g++",
+            "filledwith" : "cpp-1.3.0+commit.84983edf.Linux.g++",
+            "retesteth" : "cpp-0.0.1+commit.b346d8e6.Linux.g++",
```
Comment (chfast) [inline GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json]: What?

Comment (winsvega) [reply GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json]: retesteth commit hash

**Comment by pirapira** [comment]
@winsvega please file a PR to `testeth` repository with a new submodule pointing to this branch.  This is for running Travis.

File: GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json (hunk 1 above)
Comment (chfast) [reply GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json]: I see. But can you rename it to something more generic. E.g. "client_version" and "tool_version". And why retesteth is called "cpp"?

**Comment by holiman** [comment]
> Here is the first test filled with retesteth via RPC and custom cpp client version.
> this test should be compatible with existing cpp-develop.

How was it filled, exactly? What was used for input, and what API was used?

**Comment by winsvega** [comment]
the input is `createNameRegistratorPerTxsNotEnoughGasFiller.json`
it was filled with this RPC methods: 
https://github.com/ethereum/retesteth/issues/5

(which are not final and it's format will probably be changed)

File: GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json (hunk 1 above)
Comment (winsvega) [reply GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json]: `tool_version` does not look more generic then retesteth
`cpp` states for cpp implementation

Comment (pirapira) [reply GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json]: @winsvega can you make it, something like

```
"filling-rpc-host-version" : "cpp-1.3.0+commit.84983edf.Linux.g++",
"filling-runner-version" : "retesteth-0.0.1+commit.abdcda.Linux.g++"
```

Comment (pirapira) [reply GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json]: @winsvega let's prepare for alternative implementations of `retesteth` and let's assume not all of them are called `retesteth`.

Comment (winsvega) [reply GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json]: `filling-rpc-client`
`filling-tool-version`

Comment (chfast) [reply GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json]: @winsvega For consistency, you either add `-version` suffix or not.

Comment (pirapira) [reply GeneralStateTests/stCallCreateCallCodeTest/createNameRegistratorPerTxsNotEnoughGas.json]: @winsvega  `cpp-ethereum` is the RPC server, not the client.  `retesteth` sends requests (client) and `cpp-ethereum` answers (server).

**Comment by pirapira** [comment]
@winsvega should this be ready for merge?

**Comment by winsvega** [comment]
cpp-ethereum testeth would fail to pass this. 
because 0 reward is considered to be a touch in this tests

**Comment by pirapira** [comment]
@winsvega are you going to fix cpp-ethereum?

**Comment by winsvega** [comment]
the tests itself might be incorrect. now moving from test_addTransaction  to eth_sendRawTransaction
seems like some of this tests are invalid again. eth client reject the transaction I guess

**Comment by winsvega** [comment]
@pirapira a PR passing this branch has just been merged to cpp.
https://github.com/ethereum/cpp-ethereum/pull/5024

**Pull request merged**